src/blueapi/service/runner.py

```diff
@@ -32,51 +35,52 @@ class WorkerDispatcher:
     _state: EnvironmentResponse
 
     def __init__(
-        self, config: ApplicationConfig | None = None, use_subprocess: bool = True
+        self,
+        config: ApplicationConfig | None = None,
+        use_subprocess: bool = True,
     ) -> None:
         self._config = config or ApplicationConfig()
         self._subprocess = None
         self._use_subprocess = use_subprocess
-        self._state = EnvironmentResponse(initialized=False)
+        self._state = EnvironmentResponse(
+            initialized=False,
+        )
+
+    def reload(self):
+        """Reload the subprocess to account for any changes in python modules"""
+        self.stop()
+        self.start()
+        LOGGER.info("Runner reloaded")
 
     def start(self):
-        if self._subprocess is None and self._use_subprocess:
-            self._subprocess = Pool(initializer=_init_worker, processes=1)
-            self._subprocess.apply(
-                logging.basicConfig, kwds={"level": self._config.logging.level}
-            )
         try:
-            self.run(start_worker, [self._config])
+            if self._use_subprocess:
+                self._subprocess = Pool(initializer=_init_worker, processes=1)
+            self.run(setup, [self._config])
+            self._state = EnvironmentResponse(initialized=True)
         except Exception as e:
             self._state = EnvironmentResponse(
-                initialized=False, error_message=f"Error configuring blueapi: {e}"
+                initialized=False,
+                error_message=str(e),
             )
-            LOGGER.exception(self._state.error_message)
-            return
-        self._state = EnvironmentResponse(initialized=True)
+            LOGGER.exception(e)
 
     def stop(self):
```

Comment:
I forget whether we came to any conclusion about whether or not this made `start`/`stop` non-idempotent and whether or not we were okay with that...

Reply:
I think for now we are ok with it. There is some risk that if things are called in the wrong order e.g. if `worker()` was called before `setup()` then things might be in an inconsistent state but I think we are ok with this risk for now.